feat(openai): add 'gpt-5.1' modelID (#10211)
vercel/ai: content/providers/01-ai-sdk-providers/03-openai.mdx

- **user** _string_
  A unique identifier representing your end-user, which can help OpenAI to monitor and detect abuse. Defaults to `undefined`.

- **reasoningEffort** _'none' | 'minimal' | 'low' | 'medium' | 'high'_
  Reasoning effort for reasoning models. Defaults to `medium`. If you use `providerOptions` to set the `reasoningEffort` option, this model setting will be ignored.

<Note>
  The 'none' type for `reasoningEffort` is only available for OpenAI's GPT-5.1

>   models. Defining `reasoningEffort` as 'none' for other models will result in

Reviewer: ```suggestion
  models. Setting `reasoningEffort` to 'none' with other models will result in
```
